### PR TITLE
bug fix in smallworld.py: random_reference and lattice_reference

### LINKED ISSUE #6150: In smallworld.py: ZeroDivisionError when passing a graph without edges
This is a similar error to #6144 that comes from the use of `nx.utils.cumulative_distribution` with graphs without edges. In #6144 there is a deeper explanation of the bug.

### PR DESCRIPTION
**Pull request by paulitapb**
Closes #6150. 

Changed random_reference, and lattice_reference, now an exception is raised if the function is called with a graph with fewer than 2 edges. Also updated the documentation.

### CODE REVIEW

File: networkx/algorithms/tests/test_smallworld.py [hunk 1]
```diff
@@ -68,3 +68,10 @@ def test_omega():
 
     for o in omegas:
         assert -1 <= o <= 1
+
+
+def test_graph_no_edges():
+    G = nx.Graph()
+    G.add_nodes_from([0, 1, 2, 3])
+    pytest.raises(nx.NetworkXError, nx.random_reference, G)
+    pytest.raises(nx.NetworkXError, nx.lattice_reference, G)
```
Comment (rossbar) [inline networkx/algorithms/tests/test_smallworld.py:77]: Test looks good, just a few minor suggestions to make it even better!

```suggestion

@pytest.mark.parametrize("f", (nx.random_reference, nx.lattice_reference))
def test_graph_no_edges(f):
    G = nx.Graph()
    G.add_nodes_from([0, 1, 2, 3])
    with pytest.raises(nx.NetworkXError, match="Graph has less than 2 edges"):
        f(G)
```

File: networkx/algorithms/smallworld.py [hunk 2]
```diff
@@ -161,6 +173,8 @@ def lattice_reference(G, niter=5, D=None, connectivity=True, seed=None):
 
     if len(G) < 4:
         raise nx.NetworkXError("Graph has less than four nodes.")
+    if len(G.edges) < 2:
+        raise nx.NetworkXError("Graph has less than two edges")
```
Comment (rossbar) [inline networkx/algorithms/smallworld.py:177]: Grammatical nit - it wouldn't hurt to update the fewer-than-4-nodes exception message as well. Not a blocker though!

```suggestion
        raise nx.NetworkXError("Graph has fewer than two edges")
```

**Review by rossbar — APPROVED** [review]
This LGTM - this produces a much more informative exception for situations that already raise exceptions. Thanks @paulitapb !

**Comment by rossbar** [comment]
FWIW the errors here seem to be due to various misspellings in the exception checking - should be an easy fix!

File: networkx/algorithms/tests/test_smallworld.py [hunk 3]
```diff
@@ -68,3 +68,10 @@ def test_omega():
 
     for o in omegas:
         assert -1 <= o <= 1
+
```
Comment (SultanOrazbayev) [inline networkx/algorithms/tests/test_smallworld.py:71]:
```suggestion


```

The tests will need separation by two blank lines.

**Pull request merged**